src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebRequestPSCmdlet.Common.cs

```diff
@@ -904,8 +904,6 @@ internal virtual void PrepareSession()
                 }
                 else if (ProxyUseDefaultCredentials)
                 {
-                    // If both ProxyCredential and ProxyUseDefaultCredentials are passed,
-                    // UseDefaultCredentials will overwrite the supplied credentials.
                     webProxy.UseDefaultCredentials = true;
```

Comment:
Hmm, I think really the comment is correct since we can `iwr -UseDefaultCredentials:$false -Credentials $cred` th

Perhaps it makes sense to remove the `if`
```c#
                else
                {
                    // If both ProxyCredential and ProxyUseDefaultCredentials are passed,
                    // UseDefaultCredentials will overwrite the supplied credentials.
                    webProxy.UseDefaultCredentials = UseDefaultCredentials ;
                }
```

Reply:
No this is wrong, we have both `-UseDefaultCredentials` (for Authentication) and`-ProxyUseDefaultCredentials` (for Proxy)

Reply:
I updated my proposal above.

Also please remove last commit as unrelated the PR.

Reply:
Cases:
```ps1
iwr -Proxy $proxy
#GOOD proxy with no credentials

iwr -Proxy $proxy -ProxyCredential $credentials
#GOOD proxy with credentials

iwr -Proxy $proxy -ProxyUseDefaultCredentials:$true
#GOOD proxy with default credentials

iwr -Proxy $proxy -ProxyUseDefaultCredentials:$false
#GOOD proxy with no credentials

iwr -Proxy $proxy -ProxyCredential $credentials -ProxyUseDefaultCredentials:$true
#ERROR

iwr -Proxy $proxy -ProxyCredential $credentials -ProxyUseDefaultCredentials:$false
#GOOD but redundant proxy with credentials

iwr -ProxyCredential $credentials -ProxyUseDefaultCredentials
#ERROR no proxy
```

I think we could implement your proposal but it won't change the current behaviour and we should still remove the comment

Reply:
It seems the comment is not add critical information. So I suggest:
```c#
                else
                {
                    webProxy.UseDefaultCredentials = ProxyUseDefaultCredentials;
                }
```